tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -3034,6 +3033,73 @@ static LogicalResult LegalizeFloatingPointPrelu(Operation* op,
   return success();
 }
 
+// Support both PReLU and leaky ReLU. The computation of PReLU is very similar
+// to leaky Relu that, having a parameter `alpha` to control a slope for
+// negative values but alpha in PReLU is a learned tensor, so compared with
+// leaky ReLU, an extra element-wise multiply alpha and input is needed for
+// PReLU.
+static LogicalResult LegalizeQuantizedLeakyAndPrelu(Operation* op,
+                                                    PatternRewriter& rewriter,
+                                                    Value input,
+                                                    double alpha_scale,
+                                                    ShapedType output_type) {
+  ShapedType rescale_type = output_type.clone(rewriter.getI32Type());
+  ShapedType input_type = input.getType().dyn_cast<ShapedType>();
+
+  UniformQuantizedType input_qtype =
```

Comment:
Are these guaranteed to be UniformQuantizedType? It may be worth using `dyn_cast` instead then triggering a match error to avoid segfaulting.

Reply:
Right. Apply `dyn_cast` instead.